### ISSUE TITLE
Spurious error on checked_add

### ISSUE THREAD

**Issue by nchong-at-aws** [model-checking/kani#596]
I tried this code:

```rust
pub fn main() {
    let n : u32 = __nondet();
    let m : u32 = __nondet();
    let _result = n.checked_add(m);
}
```

using the following command line invocation:

```
rmc example.rs
```

with RMC version: ebadc4cfeb69b8c02eb6a202d0c31d22eab81a8c

I expected to see this happen: VERIFICATION SUCCESS

Instead, this happened:

```
[overflow.1] arithmetic overflow on unsigned + in var_6 + var_7: FAILURE
```
